documentation/src/test/java/org/hibernate/search/documentation/search/metamodel/MetamodelIT.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.search.documentation.search.metamodel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmUtils.with;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+
+import jakarta.persistence.EntityManagerFactory;
+
+import org.hibernate.search.documentation.search.predicate.EmbeddableGeoPoint;
+import org.hibernate.search.documentation.testsupport.BackendConfigurations;
+import org.hibernate.search.documentation.testsupport.DocumentationSetupHelper;
+import org.hibernate.search.mapper.orm.Search;
+import org.hibernate.search.mapper.orm.scope.SearchScope;
+import org.hibernate.search.mapper.orm.session.SearchSession;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class MetamodelIT {
+
+	private static final int ASIMOV_ID = 1;
+	private static final int MARTINEZ_ID = 2;
+
+	private static final int BOOK1_ID = 1;
+	private static final int BOOK2_ID = 2;
+	private static final int BOOK3_ID = 3;
+	private static final int BOOK4_ID = 4;
+
+	@RegisterExtension
+	public DocumentationSetupHelper setupHelper = DocumentationSetupHelper.withSingleBackend( BackendConfigurations.simple() );
+
+	private EntityManagerFactory entityManagerFactory;
+
+	@BeforeEach
+	void setup() {
+		entityManagerFactory = setupHelper.start().setup( Book.class, Author.class, EmbeddableGeoPoint.class );
+
+		DocumentationSetupHelper.SetupContext setupContext = setupHelper.start();
+		entityManagerFactory = setupContext.setup( Book.class, Author.class, EmbeddableGeoPoint.class );
+		initData();
+	}
+
+	@Test
+	void entryPoint() {
+		with( entityManagerFactory ).runInTransaction( entityManager -> {
+			// tag::entryPoint[]
+			SearchSession searchSession = /* ... */ // <1>
+					// end::entryPoint[]
+					Search.session( entityManager );
+			// tag::entryPoint[]
+			SearchScope<Book__, Book> scope = Book__.INDEX.scope( searchSession ); // <2>
+
+			List<Book> hits = searchSession.search( scope )
+					.where( f -> f.match()
+							.field( Book__.INDEX.title ).field( Book__.INDEX.description ) // <3>
+							.matching( "robot" ) )
+					.fetchHits( 20 );
+			// end::entryPoint[]
+			assertThat( hits )
+					.extracting( Book::getId )
+					.containsExactlyInAnyOrder( BOOK1_ID, BOOK2_ID, BOOK3_ID, BOOK4_ID );
+		} );
+	}
+
+	@Test
+	void compileCheck() {
+		withinSearchSession( searchSession -> {
+			SearchScope<Book__, Book> scope = Book__.INDEX.scope( searchSession );
```

Comment:
Not a problem here because you don't include the code in the docs, but if you do include similar code in the docs, I would encourage using `var scope`... Otherwise this feels very verbose.